frontend/src/components/metadata/CreateMetadataDefinition.tsx

```diff
@@ -0,0 +1,307 @@
+			clearForm();
+		}
+	}
+
+	const clearForm = () => {
+		// Reset stepper
+		setActiveStep(0);
+
+		// Reset state
+		setFormInput({
+			metadataName: "",
+			metadataDescription: "",
+			context: "https://schema.org/",
+			fields: [{
+				name: "",
+				list: false,
+				widgetType: "",
+				config: {
+					type: "",
+					options: ""
+				},
+				required: false,
+			}]
+		})
+	}
+
+	const handleNext = () => {
+		setActiveStep((prevActiveStep) => prevActiveStep + 1);
+	};
+
+	const handleBack = () => {
+		setActiveStep((prevActiveStep) => prevActiveStep - 1);
+	};
+
+	return (
+		<div className="outer-container">
+			<div className="inner-container">
+				<Stepper activeStep={activeStep} orientation="vertical">
+					<Step key="create-metadata">
+						<StepLabel>Create Metadata Definition*</StepLabel>
+						<StepContent>
+							<TextField
+								variant="outlined"
+								margin="normal"
+								required
+								fullWidth
+								autoFocus
+								id="metadataName"
+								label="Metadata Name"
+								placeholder="Please enter metadata name"
+								value={formInput["metadataName"]}
+								onChange={(event) => { handleInputChange(0, "metadataName", event.target.value); }}
+							/>
+							<TextField
+								variant="outlined"
+								margin="normal"
+								required
+								fullWidth
+								id="metadata-description"
+								label="Metadata Description"
+								placeholder="Please enter metadata description"
+								value={formInput.metadataDescription}
+								onChange={(event) => { handleInputChange(0, "metadataDescription", event.target.value); }}
+							/>
+							<TextField
+								variant="outlined"
+								margin="normal"
+								required
+								fullWidth
+								id="metadata-context"
+								label="Metadata Context"
+								placeholder="Please enter metadata context"
+								value={formInput.context}
+								onChange={(event) => { handleInputChange(0, "context", event.target.value); }}
+							/>
+							<Button variant="contained" onClick={handleNext}>Next</Button>
+						</StepContent>
+					</Step>
+
+					{formInput.fields.map((input, index) => {
+						return (<Step key={index}>
+								{index == 0 ? 
+									<StepLabel>Add metadata entry*
+										{index == activeStep - 1 ? 
+										<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+											<AddBoxIcon />
+										</IconButton>
+										: <></>}
+									</StepLabel>
+								:
+									<StepLabel>Add additional entry
+										{index == activeStep - 1 ? 
+										<>
+											<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+												<AddBoxIcon />
+											</IconButton>
+											<IconButton color="primary" size="small" onClick={() => removeField(index)}>
+												<DeleteOutlineIcon />
+											</IconButton>
+										</>
+										: 
+										<IconButton size="small" onClick={() => removeField(index)}>
+											<DeleteOutlineIcon />
```

Comment:
When I click this delete Icon in this state, it behaves very weird
<img width="459" alt="image" src="https://user-images.githubusercontent.com/13950475/196750339-0cab5e2f-12e3-44c9-8559-f8b7b7919674.png">

Reply:
Fixed the issue -- the stepper state wasn't being properly updated when an entry was deleted.